pkg/server/commands/list_objects.go

```diff
@@ -229,6 +229,10 @@ func (q *ListObjectsQuery) evaluate(
 					return
 				}
 
+				if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
```

Comment:
This still doesn't feel right to me. If we experience a cancellation or deadline error we should still surface upwards. It's the callers responsibility to choose what to do with it, not the callee's responsibility here.

Without this code though you get a panic because you. may try to send on the channel below meanwhile the caller has returned.

Reply:
I'm going to move the WaitGroup up above this and add the ReverseExpand bit to the WaitGroup. That should avoid the closure of the channel before this function may yield on it. I believe that is the best course of action.